src/SharpCompress/Writers/Tar/TarWriter.cs

```diff
@@ -91,7 +91,13 @@ public void Write(string filename, Stream source, DateTime? modificationTime, lo
         header.Size = realSize;
         header.Write(OutputStream);
 
-        size = source.TransferTo(OutputStream);
+        if (realSize >= int.MaxValue)
```

Comment:
why restrict the size? Wouldn't be better to keep the length to a long like in it used too. I can see a use case for a 2GB file being tar the 7zipped

Reply:
the `Write` command only accepts up to an int.  I could subtract and manage it but that's a future PR if it's a problem.

Reply:
I don't see why this is an issue. The write only takes limited amount (81920 bytes) using ReadTransferBlock. The file will be separated into parts for reading then writing to tar at a time.

Reply:
you're right...overall size has nothing to do with transfer size/array

brain fart